### DISCUSSION TITLE
BSim Elasticsearch setup - ERROR Invalid URL specified: BSim URL missing DB name/path (BSimLaunchable)

### DISCUSSION THREAD

**Discussion by itsallmathematics** [NationalSecurityAgency/ghidra#7207] [Q&A]
I don't think this is a bug; I'm probably missing a setup step here. Please see the below command and output:

```
[researcher@localhost support]$ ./bsim createdatabase elastic://localhost:9200/ medium_nosize -u "bsim_user"

openjdk version "21.0.3" 2024-04-16 LTS

OpenJDK Runtime Environment Temurin-21.0.3+9 (build 21.0.3+9-LTS)

OpenJDK 64-Bit Server VM Temurin-21.0.3+9 (build 21.0.3+9-LTS, mixed mode)

Max-Memory: 908 MBytes

ERROR Invalid URL specified: BSim URL missing DB name/path (BSimLaunchable) 
```

Do you know how I can generate the DB URL from Elasticsearch that BSim needs? I am not very familiar with Elasticsearch as a database. I am currently on Ghidra 11.1.2.

**Comment by gemesa** [comment]
You are specifying an incorrect BSim URL form: `elastic://localhost:9200/` which should be sg like `elastic://localhost:9200/my_db`. You can see this in the [usage info](https://github.com/NationalSecurityAgency/ghidra/blob/bb4853e414f0030880428059b3a8c94a817017d9/Ghidra/Features/BSim/src/main/java/ghidra/features/bsim/query/ingest/BSimLaunchable.java#L985). Btw I am using BSim with PostgreSQL but have some usage examples here: https://github.com/NationalSecurityAgency/ghidra/pull/7085#issuecomment-2438640305 (scroll past the `bsim_ctl` commands).
